### ISSUE TITLE
[Feature] k-Lipschitz BatchNorm2d

### ISSUE THREAD

**Issue by leksandrs** [deel-ai/deel-torchlip#47]
Hey!

I noticed that `BatchCentering` only applies, well, centering. How about you guys include a `LipBatchNorm` layer as well?

The constraint is described in https://arxiv.org/pdf/1804.04368, page 12.

The lipschitz constant of such a layer is max_i | gamma_i / sqrt(var_i + eps) |, where gamma is the weight of the layer, and var the running variance.

Technically, one should use the variance of the current batch, but eh, the running variance should be fine. One could also just check whether we're in `train` or `eval` mode.

**Comment by franckma31** [comment]
Two new layer have been introduced in Torchlip v1.0.5
 * BatchCentering support for multi-GPU training, based on a running_mean accumulator and a running_num_batches (implemented as a special case of batchLipNorm class)
* BatchLipNorm compute the variance on batch and full set by using running means of square. This layer is not Lipschitz and can be used with a SharedLipFactory to maintain the global Lipschitz constant
* Vanilla export for BatchCentering and BatchLipNorm with the layer ScaleBiasLayer
* LipschitzModule class inherits from ScaledLipschitzModule to be able to register the coefficient_lip to a SharedLipFactory
